### DISCUSSION TITLE
Unable to upload blink program, Error: No debug probe detected.

### DISCUSSION THREAD

**Discussion by Murugesh-Hobbyist** [stm32duino/Arduino_Core_STM32#2467] [Q&A]
I'm using the below board.
https://robu.in/product/stm32f401ccu6-minimum-system-board-microcomputer-stm32-arm-core-board/

![Screenshot - 28-07-2024 , 15_42_26](https://github.com/user-attachments/assets/59ac00d7-37bd-4984-b67d-9576bd41e4f5)

Program I tried to upload
void setup() {
  pinMode(B10, OUTPUT);
}

void loop() {
  digitalWrite(B10, HIGH);  delay(1000);
  digitalWrite(B10, LOW);   delay(1000); 
}

Resulted monitor while upload,
Sketch uses 10816 bytes (4%) of program storage space. Maximum is 262144 bytes.
Global variables use 880 bytes (1%) of dynamic memory, leaving 64656 bytes for local variables. Maximum is 65536 bytes.
      -------------------------------------------------------------------
                       STM32CubeProgrammer v2.17.0                  
      -------------------------------------------------------------------
Error: No debug probe detected.

**Expected behavior**
Upload completed

Obtained result below.
![Screenshot - 28-07-2024 , 16_46_26](https://github.com/user-attachments/assets/8a740079-aa28-478d-babd-46ac9190740d)

**Desktop (please complete the following information):**
 - OS: Win 11
 - Arduino IDE version: 2.3.2
 - STM32 core version: 2.8.1
 - Upload method: SWD

**Board (please complete the following information):**
 - Name: STM32F401CCU6 

Please excuse if its a asimple solution. I read through the forums for hours with no hints.
I registered my acccount with stm32duino forum but seems it will take time for ur team to approve my request. so there will be a delay to raise this question. so I posted here. please help.

**Comment by fpistm** [comment]
You need a STLink with swd....

**Reply by Murugesh-Hobbyist** [reply]
ok without that can't we program through arduino IDE usb port?

if not, please confirm can i go for the below module because i dont understand (with swd)
https://robokits.co.in/programmers/stm32-stm08/st-link-v2?srsltid=AfmBOoo4_15jBD33fjKG5VXDqYEF17bYOcMwSMalkQx2xBSqUXfJuq2Va0E

**Reply by fpistm** [reply]
You can use dfu but you have to enter in dfu mode before upload.

**Reply by Murugesh-Hobbyist** [reply]
can you please suggest any video link/docs for the same?
if entering dfu is an option without any additional harware, is there any reason for you to suggest stlink?

**Reply by fpistm** [reply]
No reason to suggest STLink, just you error was "Error: No debug probe detected." so I simply answered why you get this issue.
About DFU, I guess you need to press Boot 0 button before plug usb cable.

**Comment by siarapatel** [comment]
Hi I am facing the same issue with another STM32 board. I am trying to program it but I am getting the following error: 
Sketch uses 17336 bytes (6%) of program storage space. Maximum is 262144 bytes.
Global variables use 1228 bytes (1%) of dynamic memory, leaving 64308 bytes for local variables. Maximum is 65536 bytes.
Warning: long options not supported due to getopt from FreeBSD usage.
Selected interface: dfu
      -------------------------------------------------------------------
                        STM32CubeProgrammer v2.19.0                  
      -------------------------------------------------------------------

I have the STM32CubeProgrammer and I am trying to use the dfu mode to program. I am holding down the boot button as I upload the code as well. Any other suggested solutions??

**Reply by fpistm** [reply]
I see no error only a warning.
Ensure to be in dfu mode. Enter in this mode depends of your board.

**Reply by siarapatel** [reply]
Hi there, I am now getting this error: 

Sketch uses 17508 bytes (6%) of program storage space. Maximum is 262144 bytes.
Global variables use 1228 bytes (1%) of dynamic memory, leaving 64308 bytes for local variables. Maximum is 65536 bytes.
Warning: long options not supported due to getopt from FreeBSD usage.
Selected interface: serial
      -------------------------------------------------------------------
                        STM32CubeProgrammer v2.19.0                  
      -------------------------------------------------------------------

Serial Port cu.SLAB_USBtoUART is successfully opened.
Port configuration: parity = even, baudrate = 115200, data-bit = 8,
                     stop-bit = 1.0, flow-control = off

Timeout error occured while waiting for acknowledgement.
Error: Activating device: KO. Please, verify the boot mode configuration and check the serial port configuration. Reset your device then try again... 
Failed uploading: uploading error: exit status 1

**Reply by siarapatel** [reply]
Not sure why this is timing out

**Reply by fpistm** [reply]
The cube programmer wait the ack from the bootloader.
Please read this Application note: AN2606 STM32 microcontroller system memory boot mode
To know how enter in specific mode. Your issue is not related to Arduino. You can try with the STM32CubeProgrammer GUI. While you don't succeed no need to try with Arduino IDE.

**Reply by siarapatel** [reply]
Hi, ok I will read that application notes. However, I need to program with Arduino, I dont want to use the STM32CubeProgrammer

**Reply by fpistm** [reply]
Arduino call the cubeprogrammer cli....

**Comment by lohikarhu** [comment]
Just wanted to acknowledge all the support that I see "hitting my mailbox"!
A lot of work!
Merçi Beaucoup/vielen Dank/Shukran/Arrigato etc.😉 (I only know to say it
in 13 languages, but spelling! Not!)
